### PR TITLE
fetch-tool: Remove project-wide lint disables

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR updates the lint config for fetch-tool to remove project-wide disables. Most of the code changes were due to @typescript-eslint/strict-boolean-expressions.

Disabling rules project-wide is a bad practice that lets unlinted code slip through. Many of our packages, including this one, have unnecessary disables, too, for rules that aren't even being broken in the project.

### CODE REVIEW

File: packages/tools/fetch-tool/src/fluidFetchArgs.ts [hunk 1]
```diff
@@ -199,8 +199,8 @@ function checkArgs() {
         paramNumSnapshotVersions = Math.max(paramSnapshotVersionIndex + 1, paramNumSnapshotVersions);
     }
 
-    if (!paramURL) {
-        if (paramSaveDir) {
+    if (paramURL !== undefined) {
```
Comment (Josmithr) [inline packages/tools/fetch-tool/src/fluidFetchArgs.ts:202]: Should be `===`

File: packages/tools/fetch-tool/src/fluidFetchMessages.ts [hunk 2]
```diff
@@ -205,7 +205,7 @@ async function* saveOps(
         if (files.length === 0) {
             curr = firstAvailableDelta;
         }
-        if (!result.done) {
+        if (result.done === true) {
```
Comment (Josmithr) [inline packages/tools/fetch-tool/src/fluidFetchMessages.ts:208]: Should be `!==` (or explicit check for undefined | false)

File: packages/tools/fetch-tool/src/fluidFetchMessages.ts [hunk 3]
```diff
@@ -250,9 +250,9 @@ export async function fluidFetchMessages(documentService?: IDocumentService, sav
         return;
     }
 
-    const files = !saveDir ?
-        undefined :
-        fs.readdirSync(saveDir)
+    const files = saveDir === undefined
+        ? undefined
+        : fs.readdirSync(saveDir)
```
Comment (Josmithr) [inline packages/tools/fetch-tool/src/fluidFetchMessages.ts:255]: Not really related to this PR, but do you know why we're using the synchronous version of this in an async function? Should we file a bug for this?

File: packages/tools/fetch-tool/src/fluidFetchSnapshot.ts [hunk 4]
```diff
@@ -88,7 +88,7 @@ function fetchBlobs(prefix: string,
             // Use the blobIdMap to assign a number for each unique blob
             // and use it as a prefix for files to avoid case-insensitive fs
             let index = blobIdMap.get(blobId);
-            if (!index) {
+            if (index !== undefined) {
```
Comment (Josmithr) [inline packages/tools/fetch-tool/src/fluidFetchSnapshot.ts:91]: Should be `===`

**Review by Josmithr — CHANGES_REQUESTED** [review]

File: packages/tools/fetch-tool/src/fluidFetchArgs.ts (hunk 1 above)
Comment (tylerbutler) [reply packages/tools/fetch-tool/src/fluidFetchArgs.ts:202]: Oops!

File: packages/tools/fetch-tool/src/fluidFetchMessages.ts (hunk 3 above)
Comment (tylerbutler) [reply packages/tools/fetch-tool/src/fluidFetchMessages.ts:255]: #10659

File: packages/tools/fetch-tool/src/fluidFetchMessages.ts (hunk 2 above)
Comment (tylerbutler) [reply packages/tools/fetch-tool/src/fluidFetchMessages.ts:208]: Oops!

**Review by Josmithr — APPROVED** [review]

**Pull request merged**